### PR TITLE
Add counters to default summers

### PR DESCRIPTION
**Pull request by MansurAshraf**
This will not compile till new version of Algebird is pulled in but its using features from https://github.com/twitter/algebird/pull/373

### CODE REVIEW

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/BuildSummer.scala [hunk 1]
```diff
@@ -46,19 +47,27 @@ object BuildSummer {
...
     }
   }
 
-  private[this] final def legacyBuilder(storm: Storm, dag: Dag[Storm], node: StormNode) = {
+  private[this] final def legacyBuilder(storm: Storm, dag: Dag[Storm], node: StormNode, jobID: JobId) = {
     val nodeName = dag.getNodeName(node)
     val cacheSize = storm.getOrElse(dag, node, DEFAULT_FM_CACHE)
     logger.info("[{}] cacheSize lowerbound: {}", nodeName, cacheSize.lowerBound)
 
+    val memoryCounter = counter(jobID, Name(nodeName), Name("memory"))
```
Comment (ianoc) [inline summingbird-storm/src/main/scala/com/twitter/summingbird/storm/BuildSummer.scala]: Looks good, though i think a comment for each of these to describe them would be good. What does memory mean here?

**Comment by MansurAshraf** [comment]
build failing on 2.10.4 due to serialization exception on `com.twitter.summingbird.Group`. I am unable to reproduce this locally and all tests are passing on 2.10.4. 
```
Caused by: java.lang.ClassNotFoundException: com.twitter.summingbird.Group
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:274)
	at java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:625)
```

**Comment by MansurAshraf** [comment]
`Group` is on [classpath] so I doubt its a classpath issue.(https://github.com/MansurAshraf/summingbird/blob/feature/metrics/summingbird-core/src/main/scala/com/twitter/summingbird/Counter.scala#L21)

**Comment by johnynek** [comment]
This is also this failure: https://travis-ci.org/twitter/summingbird/jobs/49233461#L2698

But this failure is strange.

**Comment by isnotinvain** [comment]
I think I might just merge this into #565 since it's not going to work w/o this I think

**Pull request merged**